### ISSUE TITLE
Темная тема интерфейса

### ISSUE THREAD

**Issue by denismaster** [limonte/dear-habr#17]
Собственно, было бы очень неплохо видеть темную тему для интерфейса сайта)Поможет ночным трудягам, да и читается легче)

**Comment by Kapkozov** [comment]
Всеми руками ЗА!

**Comment by limonte** [comment]
@Kapkozov 

> Всеми руками ЗА!

1е предупреждение за неинформативный комментарий.

**Comment by limonte** [comment]
По моему скромному мнению, этот функционал не должен быть частью того или иного ресурса.
Темная тема решается с помощью сторонних инструментов:

- Stylish
- расширения, например [Dark Reader](https://chrome.google.com/webstore/detail/dark-reader/eimadpbcbfnmbkopoojfekhnkhdbieeh?hl=en)

Если я не прав, ткните пожалуйста в ресурс, который поддерживает темную тему без сторонних инструментов.

**Comment by S-Kerrigan** [comment]
Я согласен с тем, что еще ни один ресурс я не видел с темной, светлой темамим (и переключателем между ними). Но кто мы такие? IT-шники, что на острие прогресса или простые "диванные войска/офисный планктон"?
То, что этого еще никто не двинул на масс-ресрусах никак не означает, что это глупо - возможно это слишком не банально. Люди могут просто в целом не осозновать, что "а так можно?".

**Comment by jok40** [comment]
Я тоже за тёмную тему, потому что стайлиш и иже с ними не всегда есть возможность использовать. Например, при чтении сайта с мобильного дивайса.

**Comment by extempl** [comment]
Для тех, кто всё ещё ждёт, но лень было запилить своё - есть актуальный юзерстиль (https://github.com/WaveCutz/habrahabr.ru_night-mode) и юзерскрипт (с переключателем, использующий юзерстиль выше) - https://github.com/keyten/HES
И о том, и о другом были статьи на "хабрах"

**Comment by VladReshet** [comment]
limonte, на вопрос о ресурсе который поддерживает ночную тему "из коробки" - мобильная версия reddit-а. Не приложение, а просто мобильная версия сайта. Reddit вроде как ну оочень крупная площадка :)
![screenshot_1](https://cloud.githubusercontent.com/assets/7568590/20130654/ff1bd656-a660-11e6-95fb-a5b9a2264d03.jpg)
![screenshot_2](https://cloud.githubusercontent.com/assets/7568590/20130655/ff1e04bc-a660-11e6-9bb2-eab92acc1d7c.jpg)

**Comment by m0sk1t** [comment]
Надеюсь прислушаются. Поддерживаю тёмную тему, тем более встречал посты в которых кто-то уже сделал с помощью Stylish следовательно работа выполнена, осталось внедрить

**Comment by 1Boomburum** [comment]
Мы пока работаем над редизайном, тёмной темы в нём пока не предусмотрено, но будем иметь ввиду. Пока же можно использовать или юзерстили или расширения типа f.lux.

**Comment by v-sorokin** [comment]
Впереди предстоит много работы по редизайну Хабра и GT, вместе с редизайном поменяется и часть html-разметки и css-классов. Поэтому юзерстили со временем будут отваливаться и их придется постоянно переписывать, что совсем не круто. Как решение, можно попробовать вести параллельную разработку «темной темы», или стать контрибьютером для уже существующих тем. Но официально, задачи «сделать переключатель между темной и светлой темой» пока не было.

**Comment by VladReshet** [comment]
А если не секрет, используется ли при вёрстке CSS препроцессор типа Less или Saas? Если да - нельзя ли просто хранить всю цветовую схему в переменных, и потом делиться этим .less файлом? Тогда при изменении классов и стилей нужно будет просто изменить цвета в переменных.

**Comment by v-sorokin** [comment]
@VladReshet не секрет, мы используем sass препроцессор(scss), и используем переменные в том числе и для темизации. Но проблема в том, что у нас часть стилей не использует переменные, что конечно же неудобно, но поправимо. С редизайном, перепишем большую часть css-кода, заодно и сделаем что-то типа scss-конфига для темизации.

**Comment by nikulnik** [comment]
рекомендую расширение для Chrome "Deluminate"

**Comment by shpaker** [comment]
> Если я не прав, укажите пожалуйста на ресурс, который поддерживает темную тему без сторонних инструментов.

Пфф... Тьма таких. Самое удачное, что я видел быол у доки к докеру.

**Comment by iiw** [comment]
>работаем над редизайном
>тёмной темы в нём пока не предусмотрено
Это бездарно

**Comment by raiym** [comment]
stackoverflow.com тоже темную тему поддерживает

**Comment by shpaker** [comment]
> Это бездарно

@iiw кто после того как я увидел новый диз гитхаба я прям переосмыслил значение слово "бездарно"
